sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -257,59 +330,153 @@ bool SCIStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest& req, cav
+        
+
+        double stopping_accel = caseThreeSpeedProfile(stop_intersection_down_track-current_state.downtrack, current_state.speed, time_to_schedule_stop);
+        // TODO: temp val for acc- remove when carma streets is running
+        stopping_accel = 1.5;
+        resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
+          current_state.downtrack, stop_intersection_down_track-config_.stop_line_buffer, current_state.speed, crossed_lanelets[0].id(),
+          crossed_lanelets[0].id(), stopping_accel, current_state.stamp,
+          current_state.stamp + ros::Duration(time_to_schedule_stop)));
+
+        
+      }
+  }
 
-      resp.new_plan.maneuvers.push_back(composeLaneFollowingManeuverMessage(
-        case_num, current_downtrack_, stop_intersection_down_track, current_state.speed, 0.0,
-        current_state.stamp, time_to_schedule_stop,
-        lanelet::utils::transform(crossed_lanelets, [](const auto& ll) { return ll.id(); })));
-    }
-    else
-    {
-      // at the stop line
-      // stop and wait maneuver
-      auto stop_line_lanelet = nearest_stop_intersection->lanelets().front();
-      double stop_duration = (scheduled_enter_time_ - scheduled_stop_time_)*1000.0;
-      ROS_DEBUG_STREAM("Planning stop and wait maneuver");
-      resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
-        current_downtrack_, stop_intersection_down_track, current_state.speed, stop_line_lanelet.id(),
-        stop_line_lanelet.id(), current_state.stamp,
-        current_state.stamp + ros::Duration(stop_duration)));
+  bool cond1 = (int(street_msg_timestamp_) - int(scheduled_stop_time_)) > 0;
+  bool cond2 = (int(street_msg_timestamp_) - int(scheduled_enter_time_)) < 0;
+  bool time_to_stop_at_stopline = (cond1 && cond2);
+  ROS_DEBUG_STREAM("time_to_stop_at_stopline: " << time_to_stop_at_stopline);
 
-    }
+  if (time_to_stop_at_stopline)
+  {
+    base_time = std::max(scheduled_stop_time_, street_msg_timestamp_);
+    double stop_duration = (scheduled_enter_time_ - base_time)/1000;
+    ROS_DEBUG_STREAM("stop_duration:  " << stop_duration);
+    ROS_DEBUG_STREAM("Planning stop and wait maneuver");
+    double stopping_accel = 1.5;
+    auto stop_line_lanelet = nearest_stop_intersection->lanelets().front();
+    resp.new_plan.maneuvers.push_back(composeStopAndWaitManeuverMessage(
+            current_state.downtrack, stop_intersection_down_track, current_state.speed, stop_line_lanelet.id(),
+            stop_line_lanelet.id(), stopping_accel, current_state.stamp,
+            current_state.stamp + ros::Duration(stop_duration)));
+  }
+
+  if (!is_allowed_int_)
+  {
+    ROS_DEBUG_STREAM("Vehicle at stop line, waiting for access to intersection");
+    // vehicle is at the intersection but it is not allowed access, so it must wait
+    // arbitrary parameters
+    double stop_acc = 1.5;
+    double stop_duration = 99999;
```

Comment:
Recommend again parameterizing the accel, and using `std::numeric_limits<double>::max()` or `std::numer_limits<double>::infinity()` for indefinite values.

Reply:
updated